frontend/pages/SoftwarePage/components/AddPackage/AddPackage.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useContext, useEffect, useState } from "react";
+import { InjectedRouter } from "react-router";
+
+import PATHS from "router/paths";
+import { NotificationContext } from "context/notification";
+import softwareAPI from "services/entities/software";
+import { QueryParams, buildQueryStringFromParams } from "utilities/url";
+
+import AddPackageForm from "../AddPackageForm";
+import { IAddSoftwareFormData } from "../AddPackageForm/AddSoftwareForm";
+import { getErrorMessage } from "../AddSoftwareModal/helpers";
+
+const baseClass = "add-package";
+
+// 2 minutes + 15 seconds to account for extra roundtrip time.
+const UPLOAD_TIMEOUT = (2 * 60 + 15) * 1000;
+const MAX_FILE_SIZE_MB = 500;
+const MAX_FILE_SIZE_BYTES = MAX_FILE_SIZE_MB * 1024 * 1024;
```

Comment:
TODO: update this to track Martin's [PR](https://github.com/fleetdm/fleet/pull/20479)

Reply:
I will merge in the new work and make sure this is updated